lib/scout_apm/merged_layer.rb

```diff
@@ -0,0 +1,124 @@
+module ScoutApm
+  module Layers
+    # A MergedLayer is a lossy-compression approach to fall back on once we max out
+    # the number of detailed layer objects we store.  See LayerChildrenSet for the
+    # logic on when that change over happens
+    #
+    # QUESTION: What do we do if we attempt to merge an item that has children?
+    class MergedLayer
```

Comment:
I wonder if this should be named `LimitedLayer` to allow the other (future) optimization of merging layers as we accumulate them ("Oh, another of your N+1 query showed up, we'll count that in our one spot rather than making a whole new layer"), which would take the `MergedLayer` name.